### PR TITLE
refactored modal to accommodate languages better

### PR DESCRIPTION
**Pull request by dhruv1794**
@aholdahl Now modal 3.0.0 has a common structure for all the languages.  WID-385

### CODE REVIEW

File: src/classBased/modal.js [hunk 1]
```diff
@@ -9,6 +9,104 @@ class Modal {
...
+        {id: "single-feature-5", text: "Décisions d'approbation"},
+        {id: "single-feature-6", text: "instantanées"},
+        {id: "desktop-footer-1", text: "Vous n'avez qu'à choisir"},
+        {id: "desktop-footer-2", text: "Sezzle"},
+        {id: "desktop-footer-3", text: "au moment de régler"},
+        {id: "mobile-footer-1", text: "Vous n'avez qu'à choisir Sezzle"},
+        {id: "mobile-footer-2", text: "au moment de régler"},
+        {id: "terms", text: "Sous réserve d'approbation."}, 
+      ],
+    de:[
```
Comment (aholdahl) [inline src/classBased/modal.js:61]: We should have separate modal translations for the same language keys as widget translation. For language-only keys, reflect 2.0.0 translation content (interest-free  2 week intervals). For language-country keys, reflect 2.0.7 translation content (no fee, 30 day intervals).

File: src/classBased/utils.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const trackingURL = document.widgetServerBaseUrl ? `${document.widgetServerBaseUrl}/v1/event/log` : 'https://widget.sezzle.com/v1/event/log';
 const sezzleWidgetWrapperClass = 'sezzle-shopify-info-button';
+const competitorClasses = ['afterpay-parragraph', 'affirm-as-low-as',]
```
Comment (aholdahl) [inline src/classBased/utils.js:3]: `parragraph` is misspelled - should be `paragraph`

File: src/classBased/language.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  
+    _setBrowserLanguage() {
+      this._browserLanguage = navigator.language || navigator.browserLanguage || this._defaultLanguage
+    }
+  
+    getTranslation() {
+      if(this._checkIfLanguageIsValid(this._language)){
+        return this._translations[this._language];
+      }
+      return this._translations[this._defaultLanguage];
```
[the remark this answers is not in the document]
Comment (aholdahl) [reply src/classBased/language.js:27]: OR see if this works:
const northAmerica = ['US', 'CA', 'MX', 'IN', 'GU', 'PR', 'AS', 'MP', 'VI', '', null, undefined];
let lang = this.sezzleConfig.language.substring(0,2).toLowerCase();
let locale = this.sezzleConfig.language.split('-')[1];
if(northAmerica.indexOf(this._countryCode) > -1){
this.sezzleConfig.language = lang || "en"
} else if(this._checkIfLanguageIsValid(this.sezzleConfig.language)){
this.sezzleConfig.language = this.sezzleConfig.language
} else if(this._checkIfLanguageIsValid(lang+"-"+lang.toUpperCase())){
this.sezzleConfig.language = lang+"-"+lang.toUpperCase()
} else {
this.sezzleConfig.language = "en-GB"
}

File: src/classBased/utils.js (hunk 2 above)
Comment (dhruv1794) [reply src/classBased/utils.js:3]: Thanks fixed this. _Klarna_ uses shadowRoot which basically works like a react component. Will come up a way to track them and create an issue for it for the coming quarter.

File: src/classBased/language.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (aholdahl) [reply src/classBased/language.js:27]: Putting off for MVP, see my separate branch

File: src/classBased/modal.js (hunk 1 above)
Comment (aholdahl) [reply src/classBased/modal.js]: Putting off for MVP

**Review by aholdahl — APPROVED** [review]

**Pull request merged**